jetty-server/src/main/java/org/eclipse/jetty/server/SymlinkAllowedResourceAliasChecker.java

```diff
@@ -34,33 +33,42 @@
      */
     public SymlinkAllowedResourceAliasChecker(ContextHandler contextHandler)
     {
-        super(contextHandler, false);
+        super(contextHandler);
     }
 
     @Override
-    public boolean check(String uri, Resource resource)
+    public boolean check(String pathInContext, Resource resource)
     {
-        try
-        {
-            // Check the resource is allowed to be accessed.
-            if (!super.check(uri, resource))
-                return false;
+        // The existence check resolves the symlinks.
+        if (!resource.exists())
+            return false;
+
+        Path path = getPath(resource);
+        if (path == null)
+            return false;
 
-            // Approve if path is a symbolic link.
-            Path resourcePath = resource.getFile().toPath();
-            if (Files.isSymbolicLink(resourcePath))
-                return true;
+        String[] segments = pathInContext.split("/");
```

Comment:
Because we need to walk down from the top, not up from the bottom.
Why can't we split the path on '/'? We know it is a pathInContext that is always separated by '/' and there is no escape for it (because of the stupid decoding done converting URI to a path).      What is the case that this doesn't work?